cmd/fossa/report.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	logging "github.com/op/go-logging"
+	"github.com/urfave/cli"
+
+	"github.com/fossas/fossa-cli/config"
+)
+
+var reportLogger = logging.MustGetLogger("report")
+
+type dependencyResponse struct {
+	Loc struct {
+		Package  string
+		Revision string
+	}
+	Licenses []licenseResponse
+	Project  struct {
+		Title   string
+		URL     string
+		Authors []string
+	}
+}
+
+type licenseResponse struct {
+	ID       string
+	Title    string
+	FullText string
+}
+
+func reportCmd(c *cli.Context) {
+	conf, err := config.New(c)
+	if err != nil {
+		reportLogger.Fatalf("Could not load configuration: %s", err.Error())
+	}
+
+	server, err := url.Parse(conf.Endpoint)
+	if err != nil {
+		reportLogger.Fatalf("Invalid FOSSA endpoint: %s", err.Error())
+	}
+	api, err := server.Parse(fmt.Sprintf("/api/revisions/%s/dependencies", url.PathEscape(config.MakeLocator(conf.Project, conf.Revision))))
```

Comment:
This depends on `fossa upload` succeeding as well as all deep deps to be resolved.  I would imagine the implementation would query our `/api/revisions` and `/api/licenses` api.  

I prefer for the `report` command to be independent from if the user was integrated with FOSSA; and we can offer "better" data if you used a FOSSA account.

In cases where we hit stuff that hasn't been scanned, we could rely on the `fetcher.getMetadata()` responses which we'd need to build support for in FOSSA.

If we're running with this path as our v1 implementation, I would:

 1. File an issue proposing the fossa-independent report design
 2. Give user feedback that you need `fossa upload` to succeed for this command to succeed

Reply:
Will implement.